### PR TITLE
Feat/Add 'get products by IDs' job creation

### PR DESCRIPTION
**Pull request by notrueblood**
Adds a method for creating jobs to get products by IDs.

### CODE REVIEW

File: cofactr/graph.py [hunk 1]
```diff
@@ -1302,3 +1303,91 @@ def create_order(
...
+        force_refresh: bool = False,
+        schema: Optional[ProductSchemaName] = None,
+        timeout: Optional[int] = None,
+        owner_id: Optional[str] = None,
+        stale_delta: Optional[str] = None,
+        reference: Optional[str] = None,
+    ) -> List[str]:
+        """Create batch product request job.
+
+        Note: Multiple jobs are created are made if more than 250 IDs are provided.
```
Comment (josephsayad) [inline cofactr/graph.py:1320]:
```suggestion
        Note: Multiple jobs are created if more than 250 IDs are provided.
```

File: cofactr/graph.py [hunk 2]
```diff
@@ -1302,3 +1303,91 @@ def create_order(
...
+                params=drop_none_values({"owner_id": owner_id, "ref": reference}),
+                timeout=timeout,
+                follow_redirects=True,
+            )
+            res.raise_for_status()
+
+            location = res.headers.get("location")
+
+            if not location:
+                raise ValueError("No resource location found in order creation response.")
```
Comment (josephsayad) [inline cofactr/graph.py:1388]:
```suggestion
                raise ValueError("No resource location found in job creation response.")
```

File: cofactr/graph.py [hunk 3]
```diff
@@ -1302,3 +1303,91 @@ def create_order(
...
+                        }
+                        for ids_ in batched(id_batch, n=_MAX_SUB_BATCH_SIZE)
+                        if (
+                            filtering := quote(
+                                json.dumps([{"field": "id", "operator": "IN", "value": ids_}])
+                            )
+                        )
+                    ]
+                },
+                params=drop_none_values({"owner_id": owner_id, "ref": reference}),
```
Comment (josephsayad) [inline cofactr/graph.py:1379]:
```suggestion
                params=drop_none_values({"owner_id": owner_id, "reference": reference}),
```

File: cofactr/graph.py [hunk 4]
```diff
@@ -1302,3 +1303,91 @@ def create_order(
...
+                        "X-API-KEY": self.api_key,
+                    }
+                ),
+                json={
+                    "batch": [
+                        {
+                            "method": "GET",
+                            "relative_url": (
+                                f'?filtering={filtering}&'
+                                + urlencode(invariant_query_params, doseq=True)
```
Comment (josephsayad) [inline cofactr/graph.py:1368]: We can safely omit the use of `doseq`:

```suggestion
                                + urlencode(invariant_query_params)
```

**Review by josephsayad — APPROVED** [review]
Tests as expected. Be sure to bump the SDK version number and then LGTM

**Comment by notrueblood** [comment]
> Tests as expected. Be sure to bump the SDK version number and then LGTM

Thanks, I've omitted an SDK version increase because I'm publishing at the same time as #49, which bumps it for both

**Pull request merged**